src/dialogs/quick-bar/ha-quick-bar.ts

```diff
@@ -231,51 +278,50 @@ export class QuickBar extends LitElement {
     }));
   }
 
-  private _filterCommandItems(
-    items: CommandItem[],
-    filter: string
-  ): CommandItem[] {
-    return items
-      .filter(({ text }) =>
-        fuzzySequentialMatch(filter.toLowerCase(), [text.toLowerCase()])
-      )
-      .sort((itemA, itemB) => compare(itemA.text, itemB.text));
+  private async _setFilteredItems() {
+    this._filteredItems = this._commandMode
+      ? this._filterCommandItems(this._commandItems || [], this._filter)
+      : this._filterEntityItems(this._entities || [], this._filter);
   }
 
-  private _filterEntityItems(filter: string): HassEntity[] {
-    return this._entities
-      .filter(({ entity_id, attributes: { friendly_name } }) => {
-        const values = [entity_id];
-        if (friendly_name) {
-          values.push(friendly_name);
-        }
-        return fuzzySequentialMatch(filter.toLowerCase(), values);
-      })
-      .sort((entityA, entityB) =>
-        compare(entityA.entity_id, entityB.entity_id)
-      );
-  }
+  private _filterCommandItems = memoizeOne(
+    (items: CommandItem[], filter: string): CommandItem[] => {
+      return items
+        .filter(({ text }) =>
+          fuzzySequentialMatch(filter.toLowerCase(), [text.toLowerCase()])
+        )
+        .sort((itemA, itemB) => compare(itemA.text, itemB.text));
+    }
+  );
+
+  private _filterEntityItems = memoizeOne(
+    (items: HassEntity[], filter: string): HassEntity[] => {
+      return items
+        .filter(({ entity_id, attributes: { friendly_name } }) => {
+          const values = [entity_id];
+          if (friendly_name) {
+            values.push(friendly_name);
+          }
+          return fuzzySequentialMatch(filter.toLowerCase(), values);
+        })
+        .sort((entityA, entityB) =>
+          compare(entityA.entity_id, entityB.entity_id)
+        );
+    }
+  );
```

Comment:
These dont need to be a part of the class right?

Reply:
Yeah, @bramkragten has a lot of rebasing to do right now ;) This is now ancient code.

Reply:
I went ahead and did the rebasing. He's a busy guy ;)